### PR TITLE
Feature/29 convert `wl_convert_units()` to generic

### LINKED ISSUE #29: Convert `wl_convert_units` into generic
Convert `wl_convert_units` into generic to enable the possibility to use it directly on `hyperSpec` object.

### PR DESCRIPTION
**Pull request by GegznaV**
Closes #29

### CODE REVIEW

File: R/wl_convert_units.R [hunk 1]
```diff
@@ -35,6 +52,31 @@ wl_convert_units <- function(x, from, to, ref_wl = NULL) {
...
+
+#' @rdname wl_convert_units
+#' @export
+wl_convert_units.hyperSpec <- function(x, from, to, ref_wl = NULL) {
+  wl_old <- wl(x)
+  wl_new <- wl_convert_units(wl_old, from, to, ref_wl)
+  wl(x) <- wl_new
+
+  x@label$.wavelength <-
+    switch(.wl_fix_unit_name(to),
```
Comment (GegznaV) [inline R/wl_convert_units.R:67]: Create a separate function that is based on this `switch` sentence.

Comment (GegznaV) [reply R/wl_convert_units.R:67]: This will be addressed in a separate PR. See:
- #60

**Review by bryanhanson — COMMENTED** [review]
Alright, if I understand this function I see 2 items for your consideration:

1. This line `#' @param x Data for conversion.` would be clearer if it read `#' @param x Data to be converted.  Either a hyperSpec object or a numeric or integer vector.`
2. This UT ` test_that("wl_convert_units.hyperSpec works", {` actually tests both `...units.hyperSpec` and `...units.default`  You may wish to separate them.

I leave it up to you to decide if these items are worth changing.  OK to merge either way.

**Comment by GegznaV** [comment]
Thank you, @bryanhanson, for the quick review!

**Pull request merged**